MoreLinq/Delegating.cs

```diff
@@ -0,0 +1,90 @@
+#region Copyright (C) 2017 Atif Aziz. All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+#endregion
+
+// https://github.com/atifaziz/Delegating
+
+namespace Delegating
+{
+    using System;
+    using System.Threading;
+
+    static partial class Delegate
+    {
+        public static IDisposable Disposable(Action delegatee) =>
+            new DelegatingDisposable(delegatee);
+
+        public static IObservable<T> Observable<T>(Func<IObserver<T>, IDisposable> delegatee) =>
+            new DelegatingObservable<T>(delegatee);
+
+        public static IObserver<T> Observer<T>(Action<T> onNext,
+                                               Action<Exception> onError = null,
+                                               Action onCompleted = null) =>
+            new DelegatingObserver<T>(onNext, onError, onCompleted);
+    }
+
+    partial class DelegatingDisposable : IDisposable
```

Comment:
This class doesn't appear to need to be partial.

And it should probably be sealed. Same for the other Delegating classes

Reply:
These were copied verbatim from [Delegating](https://github.com/atifaziz/Delegating) and I try to maintain the original sources by as much as possible. Keeping them as partial definitions is harmless but they cannot be sealed without modifying the source code or sealing sub-classes (which seems like an overkill). Since I don't expect these to change and it's a small maintenance cost to keep the code in sync if they do change, I have gone and made these changes in 27e20d1136b1ca21878d9729474f23379fd0b6bf and 099a642622442997472d8a8f2566a8a27adbf1bc.